viewing_party/party.py

```diff
@@ -1,23 +1,131 @@
+    if num_watched == 0:
+        return 0
+    else:
+        for movie in user_data["watched"]:
+            rating_sum += movie["rating"]
+        return rating_sum/num_watched
+
+def get_most_watched_genre(user_data):
+    genre_count_dict = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    else: 
+        for movie in user_data["watched"]:
+            if not movie["genre"] in genre_count_dict:
+                genre_count_dict[movie["genre"]] = 1
+            else:
+                genre_count_dict[movie["genre"]] += 1
+    most_watched_genre = max(genre_count_dict, key=genre_count_dict.get)
+    return most_watched_genre
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def make_list_friends_watched(user_data): #helper function
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)
+    return friends_watched
+
+def get_unique_watched(user_data):
+    unique_watched = []
+    friends_watched = make_list_friends_watched(user_data)
+
+    for movie in user_data["watched"]:
+        if not movie in friends_watched:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    friends_watched = make_list_friends_watched(user_data)
+
+    for movie in friends_watched:
+        if not movie in user_data["watched"] \
+        and not movie in friends_unique_watched:
+            friends_unique_watched.append(movie)
+    return friends_unique_watched
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommendations = []
+    friends_watched = make_list_friends_watched(user_data)
+
+    for movie in friends_watched:
+        if movie not in user_data["watched"] \
+        and movie["host"] in user_data["subscriptions"] \
+        and movie not in recommendations:
+            recommendations.append(movie)
+
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    recommendations = []
+    most_watched_genre = get_most_watched_genre(user_data)
+    friends_watched = make_list_friends_watched(user_data)
+
+    for movie in friends_watched:
+        if movie["genre"] == most_watched_genre \
+        and movie not in user_data["watched"]:
+            recommendations.append(movie)
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+    recommendations = []
+    friends_watched = make_list_friends_watched(user_data)
+    
+    for favorite in user_data["favorites"]:
+        if not favorite in friends_watched:

```

Comment:
Consider writing this as `if favorite not in friends_watched:` instead to make this a little more readable.